Src/MakingSense.SafeBrowsing/GoogleSafeBrowsing/GoogleSafeBrowsingUpdater.cs

```diff
@@ -0,0 +1,171 @@
+﻿#if !(NET35 || NET40 || NETSTANDARD1_0)
+using Google.Apis.Safebrowsing.v4;
+using Google.Apis.Safebrowsing.v4.Data;
+using Google.Apis.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MakingSense.SafeBrowsing.GoogleSafeBrowsing
+{
+    /// <summary>
+    /// Allow to download Google Safe Browsing lists from HTTP and update them
+    /// </summary>
+    public class GoogleSafeBrowsingUpdater : IUpdater
+    {
+        private readonly GoogleSafeBrowsingConfiguration _configuration;
+        private readonly GoogleSafeBrowsingDatabase _database;
+        private readonly SafebrowsingService _safebrowsingService;
+
+        private const string ANY_PLATFORM = "ANY_PLATFORM";
+        private const string URL = "URL";
+        private const string RAW = "RAW";
+        private const string FULL_UPDATE = "FULL_UPDATE";
+
+        /// <summary>
+        /// Creates new instance and initializes configuration parameters
+        /// </summary>
+        /// <param name="configuration">Google Safe Browsing configuration</param>
+        public GoogleSafeBrowsingUpdater(GoogleSafeBrowsingConfiguration configuration)
+        {
+            _configuration = configuration;
+
+            _safebrowsingService = new SafebrowsingService(new BaseClientService.Initializer
+            {
+                ApiKey = _configuration.ApiKey,
+            });
+
+            _database = new GoogleSafeBrowsingDatabase();
+        }
+
+        /// <summary>
+        /// Updates Google Safe Browsing lists based on the remote resource
+        /// </summary>
+        /// <returns></returns>
+        public async Task UpdateAsync()
+        {
+            if (!_database.AllowRequest)
+            {
+                return;
+            }
+
+            var request = new FetchThreatListUpdatesRequest
```

Comment:
I am not sure how this google generated API works.  Do you know if it takes advantage of _ETag_ field?

Reply:
I don't think so - it comes null for list update requests.

Reply:
Thanks!